src/db_queries/admin_partners_students.ts

```diff
@@ -0,0 +1,124 @@
+import {Request, Response} from "express";
+import pg from "pg";
+import "moment-timezone";
+import {Conn} from "../db/conn";
+import {Helpers} from "../utils/helpers";
+import PartnerStudent from "../models/partner_student.model";
+import PartnerStudentSearch from "../models/partner_student_search.model";
+import {filterRowsBySearchParams} from "../models/partner_student.model.utils";
+
+const conn = new Conn();
+const pool = conn.pool;
+const helpers = new Helpers();
+
+
+export class AdminPartnersStudents {
+  constructor() {
+    helpers.autoBind(this);
+  }
+
+  async getAllStudentsOfOnePartner(
+    request: Request,
+    response: Response
+  ): Promise<void> {
+    const {partner_id: partnerId} = request.params;
+    const searchParameters: PartnerStudentSearch = request.body;
+
+    const client = await pool.connect();
+    try {
+      await client.query("BEGIN");
+      const students = await this.getAllStudentsOfOnePartnerFromDB(
+        partnerId,
+        searchParameters,
+        client
+      );
+      response.status(200).json(students);
+      await client.query("COMMIT");
+    } catch (error) {
+      console.log("getAllStudentsOfOnePartner", error)
+      response.status(400).send(error);
+      await client.query("ROLLBACK");
+    } finally {
+      client.release();
+    }
+  }
+
+  private async getAllStudentsOfOnePartnerFromDB(
+    partnerId: string,
+    searchParameters: PartnerStudentSearch,
+    client: pg.PoolClient
+  ) {
+    const {
+      courseFromDate,
+      courseToDate,
+    } = searchParameters;
+    let andFromToCondition = "";
+    let whereFromToCondition = "";
+    let fromToCondition = "";
+
+    if (courseFromDate || courseToDate) {
+      if (courseFromDate && courseToDate) {
+        fromToCondition = `(
+          (uc.start_date_time between '${courseFromDate}' and '${courseToDate}')
+          or (uc.start_date_time + (ct.duration * INTERVAL '1 month') between '${courseFromDate}' and '${courseToDate}')
+       )
+      `;
+      } else if (courseFromDate && !courseToDate) {
+        // If only from date is provided, we will take all courses that end after that date
+        fromToCondition = `(uc.start_date_time + (ct.duration * INTERVAL '1 month') > '${courseFromDate}')`;
+      } else if (!courseFromDate && courseToDate) {
+        // If only to date is provided, we will take all courses that begin before that date
+        fromToCondition = `(uc.start_date_time < '${courseToDate}')`;
+      }
+      andFromToCondition = ` and ${fromToCondition} `;
+      whereFromToCondition = ` where ${fromToCondition} `;
+    }
```

Comment:
I coppied this from the other implementation [here](https://github.com/MentorsWithoutBorders/mwb-connect-backend/blob/028d10aa7ea3b5cb0b85cb16e44c1a11405e3760/src/db_queries/admin_partners_mentors.ts#L53C1-L70C1)

Is this really the best way to do this? I dont know myself

Reply:
Also I think I dont understand how I am supposed to fullifil this part of the ticket, because I am not familiar with the functionality of the app in general. 

Could u please help out a bit?